src/sentry/models/artifactbundle.py

```diff
@@ -80,6 +84,17 @@ def get_ident(cls, url, dist=None):
             return sha1_text(url + "\x00\x00" + dist).hexdigest()
         return sha1_text(url).hexdigest()
 
+    @classmethod
+    def can_be_renewed(cls, date_added: datetime):
+        current_date = datetime.now(tz=pytz.UTC)
+
+        expiration_date = date_added + timedelta(days=EXPIRATION_DAYS)
+        cutoff_date = date_added + timedelta(days=EXPIRATION_DAYS + EXPIRATION_DAYS_CUTOFF)
+
+        # An artifact bundle can be renewed only if its date_added is between 90 and 97 days old. This is done in order
+        # to give users a 7 days window after 90 days from artifact creation to keep their artifact bundles alive.
+        return expiration_date <= current_date <= cutoff_date
```

Comment:
I would rather advocate for something similar to how symbolicator manages its caches:

- bump `date_added` once every 30 days if its being used actively
- delete after 90 days

Reply:
Is there any technical reason for doing so?

Reply:
either way you have two durations:
- one after which the item is eligible for refresh
- one after which the item will be deleted

the first has to be smaller than the second one, and the frequency of access has to be higher than the difference of the two. so it does not really matter after all. IMO having a hard cut after 90 days is reasonable. you can then adjust the refresh duration then based on other factors. 90/60 is reasonable as well, then you refresh half as often as with 90/30.